### PR TITLE
[kibana] support api_key

### PR DESCRIPTION
**Pull request by sachin-frayne**
## Summary

Adds support for api_key authentication in the kibana module

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally
- create an api key and grab the beats format[1]
- configure kibana module to use that api_key:
```
metricbeat.modules:
  - module: kibana
    xpack.enabled: true
    hosts: ["http://localhost:9200"]
    api_key: <key in beats format>
    period: 10s
```
- kibana metricsets are successfully collecting documents

[1]
Screenshot 2023-08-09 at 17 23 25

## Related issues

- Step one to adding api_key to Kibana module: https://github.com/elastic/enhancements/issues/22286

### CODE REVIEW

File: metricbeat/module/kibana/cluster_actions/cluster_actions.go
Comment (klacabane) [inline metricbeat/module/kibana/cluster_actions/cluster_actions.go:91]: Can we append the api key to the http clients when we init the metricsets so we don't have to worry about passing the key around ? Something similar to what the [elasticsearch module is doing](https://github.com/elatic/beats/blob/main/metricbeat/module/elasticsearch/metricset.go#L114) although I'm not sure if it's feasible here

**Review by belimawr — APPROVED** [review]

**Comment by VihasMakwana** [comment]
@sachin-frayne can you take care of CI failures. LGTM otherwise.

**Comment by VihasMakwana** [comment]
I've rebased main.

**Comment by sachin-frayne** [comment]
> I've rebased main.

Thank you

**Comment by sachin-frayne** [comment]
Hello @VihasMakwana.

I have fixed the CI, I have also changed the setting in: [.github/workflows/golangci-lint.yml](https://github.com/elastic/beats/pull/40360/files#diff-d9a956120ac84289d2650137f64bd8f0893331de05e44cc41899dd984c9e0d48) removing: 

```skip-go-installation: true```

I don't think this was causing any failures, but it debugging I found that the setting is removed/being removed: https://github.com/golangci/golangci-lint-action

Can you please review one more time?

**Comment by sachin-frayne** [comment]
Hello @VihasMakwana 

When you get a chance can you please review this?

**Comment by pierrehilbert** [comment]
@alexsapran / @pazone could you please review here?

**Comment by alexsapran** [comment]
The task is to review this line.
https://github.com/elastic/beats/pull/40360/files#diff-d9a956120ac84289d2650137f64bd8f0893331de05e44cc41899dd984c9e0d48

right @pierrehilbert?

File: .github/workflows/golangci-lint.yml
Comment (alexsapran) [inline .github/workflows/golangci-lint.yml:57]: This was deprecated in 

```
v4.0.0+ requires an explicit actions/setup-go installation step before using this action: uses: actions/setup-go@v5. The skip-go-installation option has been removed.
```

But we are running version `v1.55.2`, so I am not sure removing this achieves what you want.
Why we wan to remove this line?

**Comment by pierrehilbert** [comment]
Yes this is where your review is needed, the rest of the PR has been reviewed by @belimawr

File: .github/workflows/golangci-lint.yml
Comment (sachin-frayne) [reply .github/workflows/golangci-lint.yml:57]: I will revert this change now.

**Comment by sachin-frayne** [comment]
@pierrehilbert and @belimawr

Since I have reverted the change with the `skip-go-installation` can this move onto the next stage, I believe I am waiting for a reviewer approval.

Thank you.

File: metricbeat/module/kibana/kibana.go
Comment (belimawr) [inline metricbeat/module/kibana/kibana.go:162]: [Question]
Should this header be always set or only when `apiKey` is present (not an empty string)?

File: metricbeat/module/kibana/status/status.go
Comment (belimawr) [inline metricbeat/module/kibana/status/status.go:76]: [Question]
Same question here: should this always be set or only when `apiKey` is not empty?

Also as a suggestion:
```suggestion
	m.http.SetHeader("Authorization", "ApiKey "+m.ApiKey)
```

**Comment by sachin-frayne** [comment]
Hey @belimawr, :)

Thank you for your review and the suggestions, you are correct the header should only be set if the apikey is not an empty string, I have implemented an if statement around the function.

Please review and let me know if you are happy with the changes?

**Review by belimawr — APPROVED** [review]

**Comment by sachin-frayne** [comment]
Hello @VihasMakwana:

I am not sure, but I believe I am still waiting for your approval.

**Comment by pierrehilbert** [comment]
@sachin-frayne as @VihasMakwana and @belimawr are in the same team, one of them you should be enough. 
Looks like you need someone from @elastic/stack-monitoring

**Comment by consulthys** [comment]
@sachin-frayne is @elastic/stack-monitoring :-)

File: metricbeat/module/kibana/cluster_actions/cluster_actions.go
Comment (sachin-frayne) [reply metricbeat/module/kibana/cluster_actions/cluster_actions.go:91]: I am also not sure if it is feasible here. I tried a couple of ways, but in the end passing it through like this ended up being the easiest way for me to get it working.

**Comment by pierrehilbert** [comment]
> @sachin-frayne is @elastic/stack-monitoring :-)

I still think you need someone from your team to approve the change according to the codeowners.

**Review by klacabane — APPROVED** [review]

**Pull request merged**
